machine-learning/src/main.ts

```diff
@@ -5,7 +5,11 @@ import { Logger } from '@nestjs/common';
 async function bootstrap() {
   const app = await NestFactory.create(AppModule);
 
-  await app.listen(3003, () => {
+  const listeningPort: number = isNaN(Number(process.env.MACHINE_LEARNING_PORT))
```

Comment:
`const port = Number(process.env.MACHINE_LEARNING) || 3003`;

If we do this change, we should update `app.config.ts` to include it as optional number type.